recipe/build.sh

```diff
@@ -80,9 +80,9 @@ cmake ${CMAKE_ARGS} \
       -DENABLE_BYTERANGE=ON \
       -DENABLE_NCZARR=on \
       -DENABLE_NCZARR_ZIP=on \
-      -DENABLE_NCZARR_S3=on \
+      -DENABLE_NCZARR_S3=off \
       -DENABLE_NCZARR_S3_TESTS=off \
-      -DENABLE_S3_SDK=on \
+      -DENABLE_S3_SDK=off \
```

Comment:
My guess is that we can revert it only on the Windows builds, no?

Reply:
I have a suspicion that something is wrong with the Linux builds as well since my project's CI started to fail with a strange segfault once the latest netcdf build landed.  I am not clever enough to debug GHA and I can't reproduce locally, so I could be wildly wrong.

Reply:
I also think it's a lot less confusing for users when all platforms have the same features whenever possible.  I realize that's not a super strong argument, though, so I could go either way.

@akrherz, it seems like random segfaults are disconcerting but a little hard to tie definitively to this change, so I would be hesitant to revert these changes on those grounds.  But maybe if we revert them for consistency and you find out that the segfaults do go away, that will lend more credence to the idea that they are related.

Reply:
My again likely wrong suspicion is that the changes proposed here Unidata/netcdf-c#2741 are plaguing all platforms and not just windows.

Reply:
Okay, well it seems safest to revert for all platforms and then bring back the feature for all platforms in my view.

Reply:
Yep. Just ignore my suggestion :grimacing: 

Definitely better to revert all and re-visit this in the near future. Sorry for the distraction and feel free to merge this when you think it is ready.